tests/integration/consumer_test.rs

```diff
@@ -54,6 +57,67 @@ async fn consumer_test() {
     producer.close().await.unwrap();
 }
 
+fn hash_strategy_value_extractor(message: &Message) -> String {
+    let s = String::from_utf8(Vec::from(message.data().unwrap())).expect("Found invalid UTF-8");
+    return s;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn super_stream_consumer_test() {
+    let env = TestEnvironment::create_super_stream().await;
+    let reference: String = Faker.fake();
+
+    let message_count = 10;
+    let mut super_stream_producer = env
+        .env
+        .super_stream_producer(RoutingStrategy::HashRoutingStrategy(
+            HashRoutingMurmurStrategy {
+                routing_extractor: &hash_strategy_value_extractor,
+            },
+        ))
+        .build(&env.super_stream)
+        .await
+        .unwrap();
+
+    let super_stream_consumer: Arc<SuperStreamConsumer> = Arc::new(
+        env.env
+            .super_stream_consumer()
+            //.offset(OffsetSpecification::Next)
+            .build(&env.stream)
+            .await
+            .unwrap(),
+    );
+
+    for n in 0..message_count {
+        let msg = Message::builder().body(format!("message{}", n)).build();
+        let _ = super_stream_producer
+            .send(msg, |confirmation_status| async move {
+                println!("Message confirmed with status {:?}", confirmation_status);
+            })
+            .await
+            .unwrap();
+    }
+
+    let received_messages = Arc::new(AtomicU32::new(0));
+
+    let mut tasks = Vec::new();
+    for mut consumer in super_stream_consumer.get_consumers().await.into_iter() {
```

Comment:
@wolf4ood : This one is the most blocking issue I'm having at the moment. The basic idea that I had in mind was to make the implementation simple and create the SuperStreamConsumer as a wrapper of Consumers each one consuming from a single partition (stream). From a super_stream_consumer the caller could get the Consumers and receive the respective messages, maybe creating a thread for every Consumer as I was trying to do in this test. The issue I have is that at the moment Consumer can't be Cloneable. I tried to make it Cloneable but this is propagating some issue because: receiver: Receiver<Result<Delivery, ConsumerDeliveryError>> is not Cloneable because ConsumerDeliveryError is not Cloneable and back till the generic library Error is not Cloneable as well. I tried to let return a reference here in:  get_consumers(&self) but I have indeed this issue here:  "borrowed value does not live long enough" in this line above super_stream_consumer.get_consumers(). Wondering if you have some suggestions about that

Reply:
Probably just define Receiver<Result<Delivery, ConsumerDeliveryError>>  as  receiver: Arc<Mutex<Receiver<Result<Delivery, ConsumerDeliveryError>>>> (and fix all usages accordingly) in Consumer can allow Consumer to be Cloneable

Reply:
For the super stream consumer we could spawn the n consumers each one in a separated task

where in a while let loop we do `consumer.next()` and then send the delivery through a channel to the super stream consumer. We then we have to  implement `Stream` for the super consumer that just poll the channel receiver. 

For closing the super stream we could store in the super stream struct the `ConsumerHandle` of each spawned consumer